code/game/antagonist/antagonist_create.dm

```diff
@@ -99,11 +99,14 @@
 		player.current << "<span class='notice'>[leader_welcome_text]</span>"
 	else
 		player.current << "<span class='notice'>[welcome_text]</span>"
+	if (config.objectives_disabled)
+		player.current << "<span class='notice'>[antag_text]</span>"
 
 	if((flags & ANTAG_HAS_NUKE) && !spawned_nuke)
 		create_nuke()
 
-	show_objectives(player)
+	if (!config.objectives_disabled)
```

Comment:
Not sure I get what you're saying. Wouldn't this only mean that an antagonist wouldn't see their objectives if the config was disabled if they got objectives _before_ they became an antag?